src/utils/oauth2_util.py

```diff
@@ -76,3 +105,36 @@ def encrypt_password(password):
     hash_password = bcrypt.hashpw(password_bytes, salt)
     decode_hash_password = hash_password.decode("utf-8")
     return decode_hash_password
+
+
+def validate_roles_and_scopes(secret_key, endpoint_request):
+
+    token = endpoint_request.headers.get("access_token")
+    token_decode = jwt.decode(token, secret_key, algorithms="HS256")
+    roles = token_decode.get("roles")
+    scopes = token_decode.get("scopes")
+    secured_http_request_url = SecuredHttpRequestUrl(
+        path=str(endpoint_request.url_rule), method=endpoint_request.method
+    )
+
+    unauthorized_exception = UnAuthorizedEndpoint("Unauthorized Endpoint Access")
+    if roles and all(
```

Comment:
Could you improve readability here? the code seems confusing when we read it.

Reply:
Done.